### DISCUSSION TITLE
Sitemap blank on production only

### DISCUSSION THREAD

**Discussion by tweakitguy** [spatie/laravel-sitemap#505] [General]
Using standard config as per instructions.  Generating fine on local machine using 
`php artisan sitemap:generate`

However the same command executive on my production server (NGINX, PHP 8.1) produces an empty sitemap.  Only difference I can tell is .env pointing to https.  I have added `RequestOptions::VERIFY => false` as some have pointed out.

```html
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
<div style="all: initial !important;"/>
</urlset>

**Comment by lmduckworth** [comment]
Hi @tweakitguy, did you get anywhere with this? I'm having the same problem

**Reply by tweakitguy** [reply]
No, I removed the cron job and just using the local copy I generated.  Luckley the website doesn't allow creating new urls.

**Comment by AlexisSerneels** [comment]
Same here.

**Comment by fdjkgh580** [comment]
It's help me to fix my same problem
https://github.com/spatie/laravel-sitemap/discussions/372#discussioncomment-2774564

```php
use Spatie\Sitemap\SitemapGenerator;

$file = public_path("sitemap.xml");
$url = config("app.url");

SitemapGenerator::create($url)->configureCrawler(function ($crawler) {
    $crawler->ignoreRobots();
})->writeToFile($file);
```

![image](https://github.com/spatie/laravel-sitemap/assets/4414580/f3735840-c113-4ea0-a523-a632043d83e8)

![image](https://github.com/spatie/laravel-sitemap/assets/4414580/8aed9fa6-2f71-4274-9878-be35ce356d11)

**Reply by WMspace** [reply]
But in this case, all unwanted links that you may have noticed for the search robot as rel="nofollow" or pages with the meta tag <meta name="robots" content="noindex, nofollow" /> will be added to the sitemap.

**Comment by WMspace** [comment]
In my opinion, it is necessary to check the site (https, it does not matter) for the presence of the meta tag `<meta name="robots" content="noindex, nofollow" />` and links with `rel="nofollow"` all this will not be added to the sitemap and it is very correct.

**Comment by WMspace** [comment]
If the link cannot be closed with `rel="nofollow"`, and it is not needed in the sitemap, then you need to add: `->hasCrawled(function ( Url $url ) {...})`. Example:
```
 <a {{--rel="nofollow"--}} title="{{$lang->name}}" href="{{ route('changeLang', $lang->code) }}">
```

```
SitemapGenerator::create( config('app.url') )
            ->hasCrawled(function ( Url $url ) {
                if ( $url->segment(1) == 'changeLang' ) {
                    return;
                }
                return $url;
            })        
            ->writeToFile( public_path('sitemap.xml') );
        return ...;
```
